temporalio/lib/temporalio/worker/activity_executor/thread_pool.rb

```diff
@@ -0,0 +1,254 @@
+# frozen_string_literal: true
+
+# Much of this logic taken from
+# https://github.com/ruby-concurrency/concurrent-ruby/blob/044020f44b36930b863b930f3ee8fa1e9f750469/lib/concurrent-ruby/concurrent/executor/ruby_thread_pool_executor.rb,
+# see MIT license at
+# https://github.com/ruby-concurrency/concurrent-ruby/blob/044020f44b36930b863b930f3ee8fa1e9f750469/LICENSE.txt
+
+module Temporalio
+  class Worker
+    class ActivityExecutor
+      # Activity executor for scheduling activities in their own thread. This implementation is a stripped down form of
+      # Concurrent Ruby's `CachedThreadPool`.
+      class ThreadPool < ActivityExecutor
```

Comment:
Looks like large part of what was inherited from the original implementation is not pertinent for what we're doing. Unless you plan to make the fiber executor honor a similar API, I'd suggest trimming this one down. That's API surface and maintainance burden we don't need.

Reply:
I did trim some logic down, but the remaining API surface has more value than if hidden IMO. If it were extra logic I'd agree it adds a maintenance burden, but just exposing extra info I do not think adds much burden and I think it's valuable. It's just read-only stats (unless you're referring to something else?) and I am not too concerned with the contents of the class changing significantly to the point where we will regret exposing the stats.